func-check-gen/lib/data-access/azure-storage-helper.ts

```diff
@@ -0,0 +1,163 @@
+'use strict'
+
+import 'dotenv/config'
+import * as az from 'azure-storage'
+const QueryComparisons = az.TableUtilities.QueryComparisons
+const TableQuery = az.TableQuery
+import bluebird from 'bluebird'
+import { Logger } from '@azure/functions'
+
+export declare class AsyncTableService extends az.TableService {
+  replaceEntityAsync (table: string, entity: any): Promise<any>
+  queryEntitiesAsync (table: string, tableQuery: az.TableQuery, currentToken: az.TableService.TableContinuationToken | null): Promise<any>
+  deleteEntityAsync (table: string, entityDescriptor: any): Promise<any>
+  insertEntityAsync (table: string, entityDescriptor: unknown, options?: az.TableService.InsertEntityRequestOptions): Promise<any>
+  executeBatchAsync (table: string, batch: az.TableBatch): Promise<any>
+  deleteTableIfExistsAsync (table: string): Promise<az.ErrorOrResponse>
+}
+
+export declare class StorageError extends Error {
+  type: string
+}
+
+const preparedCheckTable = 'preparedCheck'
+let tableService: AsyncTableService
+let qService: az.QueueService
+let blobService: az.BlobService
+
+const azureStorageHelper = {
+  getFromPreparedCheckTableStorage: async function getFromPreparedCheckTableStorage (azureTableService: az.TableService, checkCode: string, logger: Logger) {
+    const query = new az.TableQuery()
+      .top(1)
+      .where(TableQuery.guidFilter('checkCode', QueryComparisons.EQUAL, checkCode))
+
+    let check
+    try {
+      const data = await tableService.queryEntitiesAsync(preparedCheckTable, query, null)
+      check = data.response.body.value[0]
+    } catch (error) {
+      const msg = `getFromPreparedCheckTableStorage(): error during retrieve for table storage check for checkCode [${checkCode}]`
+      logger.error(msg)
+      logger.error(error.message)
+      throw new Error(msg)
+    }
+
+    if (!check) {
+      const msg = `getFromPreparedCheckTableStorage(): check does not exist: [${checkCode}]`
+      logger.info(msg)
+      const error = new StorageError(msg)
+      error.type = 'NOT_FOUND'
+      throw error
+    }
+    return check
+  },
+
+  deleteFromPreparedCheckTableStorage: async function deleteFromPreparedCheckTableStorage (tableService: AsyncTableService, checkCode: string, logger: Logger) {
+    const check = await azureStorageHelper.getFromPreparedCheckTableStorage(tableService, checkCode, logger)
```

Comment:
I would also move this to the try catch block below.

Reply:
this is code that was copied over from the main functions app.  i haven't used this, nor modified it so will just delete it to keep things clear.

Reply:
this is code that was copied over from the main functions app.  i haven't used this, nor modified it so will just delete it to keep things clear.